resources/shellHelperFunctions.sh

```diff
@@ -178,11 +178,54 @@ set_npm_version () {
   npm version --allow-same-version --no-git-tag-version --no-commit-hooks "$VERSION_WITH_TAG"
 }
 
+NUM_RE='^[0-9]+$'
+
+RETRY_MAX=5
+
+# Configuration:  wait between 10 sec and 120 sec.
+
+# in seconds.
+RETRY_MAX_WAIT_RANGE=111
+# in seconds
+RETRY_MIN_WAIT=10
+
+# $1  The current retry count (optional - defaults to 0 if not a non-negative whole number)
+# $2+ (everything else) the command to retry should it fail
+reattempt_if_failing ( ) {
+  local retryCount=$1
+
+  if ! [[ "$retryCount" =~ $NUM_RE ]]; then
+    retryCount=0
+  else
+    shift
+  fi
+
+  if [[ "$retryCount" -eq "$RETRY_MAX" ]]; then
+    builder_die "Retry limit of $RETRY_MAX attempts reached."
+  fi
+
```

Comment:
```suggestion
  if (( $retryCount == $RETRY_MAX )); then
    builder_die "Retry limit of $RETRY_MAX attempts reached."
  fi

```

Uses math comparison instead of string comparison, neater